src/vm/eventpipe.cpp

```diff
@@ -374,9 +374,19 @@ void EventPipe::DisableInternal(EventPipeSessionID id, EventPipeProviderCallback
     {
         // Disable the profiler.
         SampleProfiler::Disable();
+        
+        // Get the managed command line.
+        LPCWSTR pCmdLine = GetManagedCommandLine();
+
+        // If this process started as a service we won't have a managed command line
+        if (pCmdLine == nullptr)
+        {
+            // Use the result from GetCommandLineW() instead
+            pCmdLine = GetCommandLineW();
```

Comment:
Should we be using GetCommandLineW() always? If we X sometimes and Y other times that makes it feel like there are subtle tradeoffs lurking. Can you elaborate on the difference in behavior between the APIs?

Reply:
I was talking to @AaronRobinsonMSFT with respect to this. He believes `GetCommandW()` is not the right thing to do,  We should be explicit about the fact that we don't have a command line in this case, 

One simple thing to do is to avoid calling `EventPipeEventSource::SendProcessInfo()` if the `GetManagedCommandLine()` is null. The only thing I am uncertain about that is the consumers, do we know if they are okay if the event is not available in the trace?

Reply:
@brianrob do you want the scenario was that caused you to add the ProcessInfo event to EventPipe?

@cshung - I'd suggest something minimal to prevent the AV:
  if(pCmdLine == null) pCmdLine = "";

Depending on what Brian says maybe we'd need to do more.

Reply:
Instead of fixing this in the EventPipe layer, I would instead make the API operate in a reasonable way. In this case, I would have a default value or some indication in the runtime that there is no managed command line since the runtime was started from a non `ExecuteAssembly()` entry point. I genuinely believe this isn't an EventPipe issue, but rather a failing in the runtime to properly convey how it was activated.

Reply:
Not knowing what the scenario expectations for this string just yet, null might be an acceptable default value for the runtime to provide. I think we need more info.

Reply:
>null might be an acceptable default value for the runtime to provide

I disagree with that sentiment. Usage has been entirely based around the Windows scenario and the `GetCommandLineW()` API never returns `nullptr`, which means we know that no callers expect that. I can appreciate not knowing what to provide, but historically and for the sake of consistency it shouldn't be `nullptr`. I worked with @briansull on an IBC scenario around this command line behavior and he was able to address this by building one up for linux scenarios. I doubt this is possible in this case because of how the CLR is being activated so I would simply return a string that indicates the issue and provides something to search in code (way better than scratching your head for who returned null) and wait for relevant consumer to yell at which point we can provide guidance.

Reply:
I wasn't intending to advocate for any particular implementation above  or about the impact on non-EventPipe scenarios (sorry if it sounded otherwise). I just wanted to gather info. It sounds like you already have independent reasons why you think the API should change, so I suggest we combine that with what we learn about EventPipe and then pick the best new behavior.

> I would simply return a string that indicates the issue and provides something to search in code (way better than scratching your head for who returned null)

FWIW nullptr has diagnostic benefits of its own. Sentinel is easier to root cause from a user report, but it is also much harder to detect before shipping. nullptr is the reverse - more likely to get spotted because it triggers AV, harder to root cause after you aware of it. If the current behavior had been a sentinel string, its very unlikely we would have caught the issue before shipping 3.0.

Reply:
@noahfalk Yes that is a fair counterpoint. My main gripe is the use of `nullptr` as a valid value in these kind of scenarios. The original API didn't have it and we should follow the API as closely as possible, as far as that value is concerned we can get it on most non-Windows platforms via procfs. It won't be the managed command line because there isn't one, but it will be the process command line.

The other facet of this is that by fixing this piecemeal and addressing each failure location it is possible to miss an underlying solution - i.e. just provide a consistent command line for the process. If the function starts returning `nullptr` all places will inevitably duplicate similar, not consistent, logic thus creating an inconsistent view of the command line - hence the entire reason for an API.

Reply:
> @brianrob do you want the scenario was that caused you to add the ProcessInfo event to EventPipe?
The scenario is that we didn't actually know where the trace came from.  It's pretty common to receive a trace from someone and want to make sure that the trace contains the data that you expect it to.  Up until this even, the command line was not present and so there was no way to verify what the trace contains.

Reply:
Thanks for all the responses. It appears to me that the command line is for human consumption only and it can be anything as long as it is not NULL.

Shall we come up with a consensus so that I can finish the PR?

@AaronRobinsonMSFT, @brianrob , @noahfalk 

Thank you.

Reply:
Yeah that is what it sounded like to me as well. Chatting briefly to @brianrob he said they didn't use GetCommandLineW() originally because on Linux it was producing a relatively useless sentinel string "dotnet". 

I'm fine with Andrew's current fallback behavior, or @AaronRobinsonMSFT if you wanted to change GetManagedCommandLine() so that it does a similar fallback internally and never emits nullptr that seems fine as well. There is still an ugly part of the scenario that if you run this test app on Linux we probably get the useless "dotnet" string, but it seems reasonable to file a separate lower priority issue for that and keep this immediate issue for ensuring that we do not AV. Thoughts?

Reply:
One other request : ) Whatever we implement, please add some comments that capture some of this discussion. There is a fair amount of interesting background knowledge about GetManagedCommandLine() and GetCommandLineW() in this issue and it would be useful to preserve this in the code for whenever the next set of devs look at it.